src/librbd/image/RefreshRequest.cc

```diff
@@ -1079,11 +1122,16 @@ void RefreshRequest<I>::apply() {
       m_image_ctx.order = m_order;
       m_image_ctx.features = 0;
       m_image_ctx.flags = 0;
+      m_image_ctx.op_features = 0;
+      m_image_ctx.operations_disabled = false;
       m_image_ctx.object_prefix = std::move(m_object_prefix);
       m_image_ctx.init_layout();
     } else {
       m_image_ctx.features = m_features;
       m_image_ctx.flags = m_flags;
+      m_image_ctx.op_features = m_op_features;
+      m_image_ctx.operations_disabled = (
+        (m_op_features & ~RBD_OPERATION_FEATURES_ALL) != 0ULL);
```

Comment:
@dillaman I am not sure I quite understand how it is going to be used. This looks like it should have been explained in the commit 3fe7a2a3 log message, which looks like truncated.

So `operations_disabled` is true, when m_op_features contains an unknown bit, i.e. we are an older client.  Then we can't do any operations? Probably, I will need just to wait for the next PR to see a use case )

Reply:
Sorry, I fixed the truncated commit message. The goal is to avoid introducing individual feature bits that break older clients when the older clients can still safely read and write to the image but they won't know how to properly handle some new feature. The two immediate examples will be (1) cloning v2 where snapshots will be unprotected so we cannot let an older client delete a clone that uses the new format and (2) image groups where we cannot let images be removed when they are part of a group w/o cleaning up.